erdjs/src/smartcontracts/interaction.dev.net.spec.ts

```diff
@@ -0,0 +1,167 @@
+import { StrictChecker } from "./strictChecker";
+import { DefaultInteractionRunner } from "./defaultRunner";
+import { SmartContract } from "./smartContract";
+import { BigUIntValue, OptionValue, TypedValue, U32Value } from "./typesystem";
+import { getDevnetProvider, loadAbiRegistry, loadContractCode, TestWallets } from "../testutils";
+import { SmartContractAbi } from "./abi";
+import { assert } from "chai";
+import { Interaction } from "./interaction";
+import { GasLimit } from "../networkParams";
+import { ReturnCode } from "./returnCode";
+import { Balance } from "../balance";
+import BigNumber from "bignumber.js";
+import { NetworkConfig } from "../networkConfig";
+import { Account } from "../account";
+import { BytesValue } from "./typesystem/bytes";
+
+
+describe("test smart contract interactor", function () {
+    let wallets = new TestWallets();
+    let checker = new StrictChecker();
+    let provider = getDevnetProvider();
+    let alice = new Account(wallets.alice.address);
+    let aliceSigner = wallets.alice.signer;
+    let runner = new DefaultInteractionRunner(checker, aliceSigner, provider);
+
+    it("should interact with 'answer' (local testnet)", async function () {
+        this.timeout(60000);
+
+        let abiRegistry = await loadAbiRegistry(["src/testdata/answer.abi.json"]);
+        let abi = new SmartContractAbi(abiRegistry, ["answer"]);
+        let contract = new SmartContract({ abi: abi });
+
+        // Currently, this has to be called before creating any Interaction objects, 
+        // because the Transaction objects created under the hood point to the "default" NetworkConfig.
+        await NetworkConfig.getDefault().sync(provider);
+        await alice.sync(provider);
+        await deploy(contract, "src/testdata/answer.wasm", new GasLimit(3000000), []);
+
+        let interaction = <Interaction>contract.methods.getUltimateAnswer().withGasLimit(new GasLimit(3000000));
```

Comment:
Here we've forced a cast from `any` (the type of `contract.methods.getUltimateAnswer()`) to `Interaction`, so that we could call `withGasLimit` afterwards.